src/browser/screen.js

```diff
@@ -487,20 +507,52 @@ function ScreenAdapter(screen_container, bus)
 
     this.update_buffer = function(min, max)
     {
-        if(max < min)
+        if(max < min && !layers_has_changed)
         {
             return;
         }
 
-        var min_y = min / graphical_mode_width | 0;
-        var max_y = max / graphical_mode_width | 0;
+        layers_has_changed = false;
 
-        graphic_context.putImageData(
-            graphic_image_data,
-            0, 0,
-            0, min_y,
-            graphical_mode_width, max_y - min_y + 1
-        );
+        this.clear_screen();
```

Comment:
> Is this needed?

Because we currently draw layers that fill the entire screen, probably not. I'll have a check to see if removing `clear_screen()` impacts any test programs. There is a case where we want to blank the screen:

https://github.com/ErnWong/v86/blob/932bdecf63ff2ed81e5b6259bf20dafd869d9c5d/src/vga.js#L1242-L1249

But I think we can implement a 'clear-screen' bus message instead of clearing the screen consistently.

> It certainly shouldn't be called in svga mode.

I agree.

Reply:
a `clear-screen` message sounds reasonable if we need it, but I believe in this case we could just check if `layers` is empty in this function.

Reply:
If each layer object represents a patch of graphics to be drawn on top of the existing canvas, I think an empty `layers` array would make more sense to represent a "don't draw over the existing canvas, and keep whatever's currently drawn" instead of a "clear everything". What do you think?

Reply:
Yeah, that makes sense.